src/mbgl/tile/geometry_tile.cpp

```diff
@@ -139,7 +139,7 @@ void GeometryTile::queryRenderedFeatures(
                         queryGeometry,
                         transformState.getAngle(),
                         util::tileSize * id.overscaleFactor(),
-                        std::pow(2, transformState.getZoom() - id.overscaledZ),
+                        transformState.zoomScale(transformState.getZoomFraction()),
```

Comment:
Yes, I guess I didn't account for overscaled zoom on my tests. We can revisit it again with proper unit tests to verify later.